Backend/Tumultu/Application/Common/Interfaces/IRepository.cs

```diff
@@ -1,14 +1,22 @@
-﻿using Tumultu.Domain.Common;
+using Tumultu.Domain.Common;
 
-namespace Tumultu.Application.Common.Interfaces
+namespace Tumultu.Application.Common.Interfaces;
+
+public interface IRepository<TEntity, TId> where TEntity : BaseEntity<TId>
 {
-    public interface IRepository<TEntity, TKey> where TEntity : BaseEntity<TKey> 
-    {
-        Task<ICollection<TEntity>> GetAllAsync();
-        Task<TEntity> GetByIdAsync(TKey id);
-        Task<TEntity> AddAsync(TEntity entity);
-        Task<TEntity> UpdateAsync(TEntity entity);
-        Task<TEntity> DeleteAsync(TKey id);
-        Task<TEntity> FindAsync(TKey id);
-    }
-}
+    Task<IEnumerable<TEntity>> GetAllAsync();
+
+    Task<TEntity?> GetByIdAsync(TId id);
+    
+    void Insert(TEntity entity);
+
+    void InsertAll(IEnumerable<TEntity> entities);
```

Comment:
On _InsertAll_ and _DeleteAll_ methods I would change naming convention, for example to _InsertMany_, _DeleteMany_.
Nothing is really wrong about this,  I just thought that may be a little bit more intuitive.